src/Node/Stream.purs

```diff
@@ -61,6 +66,16 @@ type Writable r = Stream (write :: Write | r)
 -- | A duplex (readable _and_ writable stream)
 type Duplex = Stream (read :: Read, write :: Write)
 
+type Data = Either String Buffer
+
+-- | Read a chunk of data into `Data`
+-- | This is a mere internal plumbing function used by functions reading data
+-- | from a stream.
+type ChunkReader = Foreign -> Data
+readChunk :: ChunkReader
+readChunk = readChunkImpl Left Right
+foreign import readChunkImpl :: (String -> Data) -> (Buffer -> Data) -> ChunkReader
```

Comment:
Small thing, but I like to make the arguments to the foreign import so that it's obvious that `Left` and `Right` should go in here, like I did previously with `onDataEitherImpl`:

```purescript
 foreign import onDataEitherImpl :: forall w eff. (forall l r. l -> Either l r) -> (forall l r. r -> Either l r) -> [...]
```